tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/core/planner.h

```diff
@@ -39,6 +39,14 @@ namespace tesseract_motion_planners
 class MotionPlanner
 {
 public:
+  /** @brief Enumeration of the types of collision checks that can be performed on a planned trajectory */
+  enum class CollisionCheckType
+  {
+    NONE = 0,
+    DISCRETE = 1,
+    CONTINUOUS = 2
+  };
```

Comment:
Do you think we should add single timestep option?

Reply:
What would a single timestep option mean in the context of collision checking a planned trajectory?

Reply:
It would just check the individual waypoint and would not interpolate using longest valid segment.

Single Timestep - check only joint states in trajectory
Discrete Continuous - Interpolates based on longest valid segment and performs discrete collision check
Cast Continuous - Interpolates based on longest valid segment and performs a cast collision check.